packages/flutter/lib/src/material/elevated_button_theme.dart

```diff
@@ -0,0 +1,160 @@
+  static ElevatedButtonThemeData lerp(ElevatedButtonThemeData a, ElevatedButtonThemeData b, double t) {
+    assert (t != null);
+    if (a == null && b == null)
+      return null;
+    return ElevatedButtonThemeData(
+      style: ButtonStyle.lerp(a?.style, b?.style, t),
+    );
+  }
+
+  @override
+  int get hashCode {
+    return style.hashCode;
+  }
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other))
+      return true;
+    if (other.runtimeType != runtimeType)
+      return false;
+    return other is ElevatedButtonThemeData && other.style == style;
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(DiagnosticsProperty<ButtonStyle>('style', style, defaultValue: null));
+  }
+}
+
+/// Overrides the default [ButtonStyle] of its [ElevatedButton] descendants.
+///
+/// See also:
+///
+///  * [ElevatedButtonThemeData], which is used to configure this theme.
+///  * [ElevatedButton.defaultStyleOf], which returns the default [ButtonStyle]
+///    for elevated buttons.
+///  * [ElevatedButton.styleOf], which converts simple values into a
+///    [ButtonStyle] that's consistent with [ElevatedButton]'s defaults.
+///  * [ThemeData.ElevatedButtonTheme], which can be used to override the default
+///    [ButtonStyle] for [ElevatedButton]s below the overall [Theme].
+class ElevatedButtonTheme extends InheritedTheme {
+  /// Create a [ElevatedButtonTheme].
+  ///
+  /// The [data] parameter must not be null.
+  const ElevatedButtonTheme({
+    Key key,
+    @required this.data,
+    Widget child,
+  }) : assert(data != null), super(key: key, child: child);
+
+  /// The configuration of this theme.
+  final ElevatedButtonThemeData data;
+
+  /// The closest instance of this class that encloses the given context.
+  ///
+  /// If there is no enclosing [ElevatedButtonsTheme] widget, then
+  /// [ThemeData.ElevatedButtonTheme] is used.
+  ///
+  /// Typical usage is as follows:
+  ///
+  /// ```dart
+  /// ElevatedButtonTheme theme = ElevatedButtonTheme.of(context);
+  /// ```
+  static ElevatedButtonThemeData of(BuildContext context) {
+    final ElevatedButtonTheme buttonTheme = context.dependOnInheritedWidgetOfExactType<ElevatedButtonTheme>();
+    return buttonTheme?.data ?? Theme.of(context).elevatedButtonTheme;
+  }
+
+  @override
+  Widget wrap(BuildContext context, Widget child) {
+    final ElevatedButtonTheme ancestorTheme = context.findAncestorWidgetOfExactType<ElevatedButtonTheme>();
+    return identical(this, ancestorTheme) ? child : ElevatedButtonTheme(data: data, child: child);
+  }
+
+  @override
+  bool updateShouldNotify(ElevatedButtonTheme oldWidget) => data != oldWidget.data;
+}
+
+
+/// Please use [ElevatedButtonTheme].
+@Deprecated(
+  'This class was briefly released with the wrong name. '
+  'The correct name is ContainedButtonThemeData. '
```

Comment:
Shouldn't this be `ElevatedButtonThemeData`?

Reply:
Yes, you're correct, thanks for pointing that out.